coders/uhdr.c

```diff
@@ -607,20 +652,33 @@ static MagickBooleanType WriteUHDRImage(const ImageInfo *image_info,
       hdrImgDescriptor.range = UHDR_CR_UNSPECIFIED;
       hdrImgDescriptor.w = image->columns;
       hdrImgDescriptor.h = image->rows;
-      hdrImgDescriptor.planes[UHDR_PLANE_Y] = yBuffer;
-      cbBuffer = ((unsigned short *) yBuffer) + aligned_width * aligned_height;
-      crBuffer = ((unsigned short *) cbBuffer) + 1;
-      hdrImgDescriptor.planes[UHDR_PLANE_UV] = cbBuffer;
-      hdrImgDescriptor.planes[UHDR_PLANE_V] = NULL;
-      hdrImgDescriptor.stride[UHDR_PLANE_Y] = aligned_width;
-      hdrImgDescriptor.stride[UHDR_PLANE_UV] = aligned_width;
-      hdrImgDescriptor.stride[UHDR_PLANE_V] = 0;
 
       hdr_profile = GetExifProfile(image, exception);
     }
     else if (image->depth == 8)
     {
-      sdrImgDescriptor.fmt = UHDR_IMG_FMT_12bppYCbCr420;
+      if (image->colorspace == YCbCrColorspace)
+      {
+        sdrImgDescriptor.fmt = UHDR_IMG_FMT_12bppYCbCr420;
```

Comment:
This can be moved to a method that accepts ` uhdr_raw_image_t` so we don't need to duplicate this?

Reply:
Resolved it in recent commit.